### PR TITLE
update transformers imports for `deepspeed` and `is_torch_xla_available`

### PR DESCRIPTION
**Pull request by Rohan138**
# What does this PR do?
Minor fixes corresponding to https://github.com/huggingface/transformers/pull/25599 and https://github.com/huggingface/transformers/pull/29334

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/onnxruntime/trainer.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
 from torch.utils.data import Dataset, RandomSampler
 from transformers.data.data_collator import DataCollator
 from transformers.debug_utils import DebugOption, DebugUnderflowOverflow
-from transformers.deepspeed import deepspeed_init, deepspeed_load_checkpoint, is_deepspeed_zero3_enabled
+from transformers.integrations.deepspeed import deepspeed_init, deepspeed_load_checkpoint, is_deepspeed_zero3_enabled
```
Comment (IlyasMoutawwakil) [inline optimum/onnxruntime/trainer.py:58]: can you add a transformers version check, for compatibility with previous versions

File: optimum/onnxruntime/trainer_seq2seq.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import torch
 from torch import nn
 from torch.utils.data import Dataset
-from transformers.deepspeed import is_deepspeed_zero3_enabled
+from transformers.integrations.deepspeed import is_deepspeed_zero3_enabled
```
Comment (IlyasMoutawwakil) [inline optimum/onnxruntime/trainer_seq2seq.py:22]: same

**Comment by Rohan138** [comment]
Resolved to check transformers versions. I also added another deprecation fix for `is_torch_tpu/xla_available` along the same lines.

**Comment by IlyasMoutawwakil** [comment]
great thanks ! can you use this method instead https://github.com/huggingface/optimum/blob/c0d9111775709aac6f1451e59911abe36b4b6c37/optimum/utils/import_utils.py#L162

**Pull request merged**